js/src/scene.ts

```diff
@@ -0,0 +1,436 @@
+// Copright (c) 2015-2020, Exa Analytics Development Team
+// Distributed under the terms of the Apache License 2.0
+/* """
+=================
+scene.ts
+=================
+A 3D scene for exatomic
+
+*/
+
+import * as three from 'three'
+import * as TrackBallControls from 'three-trackballcontrols'
+
+import * as util from './util'
+
+import { DOMWidgetModel, DOMWidgetView } from '@jupyter-widgets/base'
+import { version } from '../package.json'
+
+const semver = `^${version}`
+
+export class SceneModel extends DOMWidgetModel {
+    defaults(): any {
+        return {
+            ...super.defaults(),
+            _model_name: 'SceneModel',
+            _view_name: 'SceneView',
+            _model_module_version: semver,
+            _view_module_version: semver,
+            _model_module: 'exatomic',
+            _view_module: 'exatomic',
+            width: 200,
+            height: 200,
+        }
+    }
+}
+
+export class SceneView extends DOMWidgetView {
+    scene: three.Scene
+
+    camera: three.PerspectiveCamera
+
+    renderer: three.WebGLRenderer | null
+
+    controls: any // TrackBallControls
+
+    raycaster: three.Raycaster
+
+    mouse: three.Vector2
+
+    hudscene: three.Scene
+
+    hudcamera: three.OrthographicCamera
+
+    hudcanvas: HTMLCanvasElement
+
+    hudcontext: CanvasRenderingContext2D | null
+
+    hudtexture: three.Texture
+
+    hudsprite: three.Sprite
+
+    hudfontsize: number
+
+    promises: Promise<any>
+
+    selected: three.Mesh[]
+
+    initialize(parameters: any): void {
+        super.initialize(parameters)
+        this.initListeners()
+        this.selected = []
+        this.hudfontsize = 24
+        this.promises = this.init()
+        this.displayed.then(() => {
+            this.resize()
+            this.setCameraFromScene()
+        })
+    }
+
+    init(): Promise<any> {
+        /* """
+        init
+        ---------------
+        Promise it will work
+        */
+        window.addEventListener('resize', this.resize.bind(this))
+        return Promise.all([
+            this.initScene(),
+            this.initRenderer(),
+            this.initHud(),
+        ]).then(() => {
+            this.initObj()
+            this.finalizeHudcanvas()
+            this.finalizeInteractive()
+        })
+    }
+
+    initScene() {
+        /* """
+        initScene
+        ---------------
+        An opinionated three.js Scene 
+        */
+        const scene = new three.Scene()
+        const amlight = new three.AmbientLight(0xdddddd, 0.5)
+        const dlight0 = new three.DirectionalLight(0xdddddd, 0.3)
+        const sunlight = new three.SpotLight(0xdddddd, 0.3, 0, Math.PI / 2)
+        const shadowcam = new three.PerspectiveCamera(30, 1, 1500, 5000)
```

Comment:
Do we scale input from the `Universe` to "fit within" these defaults? If not, would it be possible to scale these values off of the dimensions of the universe (regardless of the "units" used by the `Universe` (same below for lighting).

Reply:
We do no scaling so it makes sense to adjust these passive scene participants with respect to the contents of the scene. I want to keep the details of a `Universe` separate from the `Scene`, but there is no reason the `Scene` should not adapt to its contents.

Reply:
Thanks for the answer. I'm not looking to scope creep, just understand so I'm resolving this.